src/Features/CSharp/Portable/AddObsoleteAttribute/CSharpAddObsoleteAttributeCodeFixProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using Microsoft.CodeAnalysis.AddObsoleteAttribute;
+using Microsoft.CodeAnalysis.CodeFixes;
+
+namespace Microsoft.CodeAnalysis.CSharp.AddObsoleteAttribute
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpAddObsoleteAttributeCodeFixProvider)), Shared]
+    internal class CSharpAddObsoleteAttributeCodeFixProvider
+        : AbstractAddObsoleteAttributeCodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(
+                "CS0612", //  'C' is obsolete 
+                "CS0618", //  'C' is obsolete (msg)
+                "CS0672"  // Member 'D.F()' overrides obsolete member 'C.F()'
```

Comment:
:memo: I am *way* more comfortable with CS0672 here than I am with the earlier ones. I don't want it to be easy to bypass things like this:

https://github.com/dotnet/roslyn/blob/c4189de08b943f87d9f03d45ef01e615800e359d/src/Workspaces/Core/Portable/Shared/TestHooks/IAsynchronousOperationListenerProvider.cs#L81-L82

Reply:
That's obsolete-error.  That can't be bypassed.

Reply:
Sure it can. Other obsolete code is allowed to call it. 😄

Reply:
Sorry.  I should have been more clear.  The compilers report a different error in those cases.  This is just for when you are told in a warning that you're calling something obsolete.   I view offering thsi as just a shortcut if that's how the user wants to deal with the issue.  It's up to them and their team (if they're on one) if that's a suitable approach here, or if they need to go some other way.

We don't offer this on 'Obsolete(error: true)' cases.

Reply:
:memo: It looks like this is correct. The `error: true` cases are reported as CS0619 instead of CS0618.

❓ Do we want to handle collection add ones (specifically the two warning cases)?

https://github.com/dotnet/roslyn/blob/ca2f6ad9ab68518344c2d3e25c0731c83d136701/src/Compilers/CSharp/Portable/Errors/ErrorCode.cs#L643-L645

Reply:
The collectoin cases just felt so rare, i didn't care to bother.